### PR TITLE
Create stable channel and 1.0.0 release

### PR DESCRIPTION
**Pull request by leifmadsen**
Create a new channel called stable which will host the 1.0.0
release of Smart Gateway Operator.

### CODE REVIEW

File: deploy/role.yaml [hunk 1]
```diff
@@ -23,9 +23,9 @@ rules:
   verbs:
   - get
 - apiGroups:
-  - apps.openshift.io
+  - extensions
   resources:
-  - deploymentconfigs
+  - deployments
```
Comment (pleimer) [inline deploy/role.yaml]: Ah NOW it uses deployments instead of dcs

File: roles/smartgateway/defaults/main.yml [hunk 2]
```diff
@@ -1,7 +1,7 @@
 ---
 # defaults file for smartgateway
 size: 1
-container_image_path: quay.io/infrawatch/smart-gateway:latest
```
Comment (pleimer) [inline roles/smartgateway/defaults/main.yml]: Is there now default image used?

Comment (leifmadsen) [reply roles/smartgateway/defaults/main.yml]: Override is defined here for the CSV:  https://github.com/infrawatch/smart-gateway-operator/pull/44/files#diff-cf39b9e07be6d20fbef91685302e38cdR192-R193

Default is now in the task setup itself:  https://github.com/infrawatch/smart-gateway-operator/pull/44/files#diff-e46d2eb07eed6f2d5429e475cafe6e63R7-R9

Comment (pleimer) [reply roles/smartgateway/defaults/main.yml]: Ah missed that, thanks

**Comment by leifmadsen** [comment]
# Testing

I ran through the documentation provided here to instantiate a Smart Gateway (and dependencies) in the `default` namespace, and everything comes up as I would expect.

```
oc get all
NAME                                              READY   STATUS    RESTARTS   AGE
pod/amq-interconnect-7cb9b8cb7c-8zcf8             1/1     Running   0          51m
pod/amq-interconnect-7cb9b8cb7c-kk6dc             1/1     Running   0          51m
pod/cloud1-metrics-smartgateway-96c9f69b4-qxfm4   1/1     Running   0          19m
pod/interconnect-operator-68cf895f89-w7kzx        1/1     Running   0          52m
pod/smart-gateway-operator-5d7dbf7b4c-ghfdz       2/2     Running   0          20m

NAME                                     TYPE           CLUSTER-IP       EXTERNAL-IP                            PORT(S)                                 AGE
service/amq-interconnect                 ClusterIP      172.30.199.222   <none>                                 5672/TCP,8080/TCP,55672/TCP,45672/TCP   51m
service/cloud1-metrics-smartgateway      ClusterIP      172.30.16.14     <none>                                 8081/TCP                                50m
service/kubernetes                       ClusterIP      172.30.0.1       <none>                                 443/TCP                                 28d
service/openshift                        ExternalName   <none>           kubernetes.default.svc.cluster.local   <none>                                  28d
service/smart-gateway-operator-metrics   ClusterIP      172.30.137.136   <none>                                 8686/TCP,8383/TCP                       54m

NAME                                          READY   UP-TO-DATE   AVAILABLE   AGE
deployment.apps/amq-interconnect              2/2     2            2           51m
deployment.apps/cloud1-metrics-smartgateway   1/1     1            1           50m
deployment.apps/interconnect-operator         1/1     1            1           52m
deployment.apps/smart-gateway-operator        1/1     1            1           54m

NAME                                                     DESIRED   CURRENT   READY   AGE
replicaset.apps/amq-interconnect-7cb9b8cb7c              2         2         2       51m
replicaset.apps/cloud1-metrics-smartgateway-559b944887   0         0         0       26m
replicaset.apps/cloud1-metrics-smartgateway-96c9f69b4    1         1         1       50m
replicaset.apps/interconnect-operator-68cf895f89         1         1         1       52m
replicaset.apps/smart-gateway-operator-5d7dbf7b4c        1         1         1       54m

NAME                                                    IMAGE REPOSITORY                                                                         TAGS     UPDATED
imagestream.image.openshift.io/smart-gateway-operator   default-route-openshift-image-registry.apps-crc.testing/default/smart-gateway-operator   latest   20 minutes ago

NAME                                             HOST/PORT                                        PATH   SERVICES           PORT   TERMINATION     WILDCARD
route.route.openshift.io/amq-interconnect-8080   amq-interconnect-8080-default.apps-crc.testing          amq-interconnect   8080   edge/Redirect   None
```

I also did some tests with the `SMARTGATEWAY_IMAGE` environment variable in the `1.0.0` CSV file which allows us to tag a specific Smart Gateway image to a particular CSV version. This seems to be the pattern other Operators are following, and hopefully puts us in a better position for downstream work (which seems to require images reference SHA for the tags, and not a named tag like `latest` or `v2.0.0-beta1`, but I'm not worrying about that for upstream at the moment).

I validated that if `SMARTGATEWAY_IMAGE` is not set on the SGO container at start up, that the default value of `quay.io/infrawatch/smart-gateway:latest` is used.

## Operator Scorecard

```
operator-sdk scorecard 
INFO[0000] Using config file: /home/sysop/src/github.com/infrawatch/smart-gateway-operator/.osdk-scorecard.yaml 
basic:
	Spec Block Exists                   : pass
	Labels: 
		"necessity":"required"
		"suite":"basic"
		"test":"checkspectest"

	Status Block Exists                 : pass
	Labels: 
		"necessity":"required"
		"suite":"basic"
		"test":"checkstatustest"

	Writing into CRs has an effect      : pass
	Labels: 
		"test":"writingintocrshaseffecttest"
		"necessity":"required"
		"suite":"basic"

olm:
	Provided APIs have validation       : pass
	Labels: 
		"suite":"olm"
		"test":"crdshavevalidationtest"
		"necessity":"required"

	Owned CRDs have resources listed    : pass
	Labels: 
		"suite":"olm"
		"test":"crdshaveresourcestest"
		"necessity":"required"

	Spec fields with descriptors        : pass
	Labels: 
		"suite":"olm"
		"test":"specdescriptorstest"
		"necessity":"required"

	Status fields with descriptors      : pass
	Labels: 
		"necessity":"required"
		"suite":"olm"
		"test":"statusdescriptorstest"

	Bundle Validation Test              : pass
	Labels: 
		"necessity":"required"
		"suite":"olm"
		"test":"bundlevalidationtest"
	Log:
		time="2020-03-13T11:35:24-04:00" level=info msg="loading Bundles" dir=deploy/olm-catalog/smart-gateway-operator/1.0.0/metadata
time="2020-03-13T11:35:24-04:00" level=info msg=directory dir=deploy/olm-catalog/smart-gateway-operator/1.0.0/metadata file=metadata load=bundles
time="2020-03-13T11:35:24-04:00" level=info msg="loading Packages and Entries" dir=deploy/olm-catalog/smart-gateway-operator/1.0.0/metadata
time="2020-03-13T11:35:24-04:00" level=info msg=directory dir=deploy/olm-catalog/smart-gateway-operator/1.0.0/metadata file=metadata load=package
```

File: deploy/olm-catalog/smart-gateway-operator/1.0.0/smartgateway.infra.watch_smartgateways_crd.yaml [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    kind: SmartGateway
+    listKind: SmartGatewayList
+    plural: smartgateways
+    singular: smartgateway
+  scope: Namespaced
+  version: v2alpha1
+  subresources:
+    status: {}
+  versions:
+  - name: v2alpha1
```
Comment (pkilambi) [inline deploy/olm-catalog/smart-gateway-operator/1.0.0/smartgateway.infra.watch_smartgateways_crd.yaml]: Should this still be v2alpha1 or stable?

Comment (leifmadsen) [reply deploy/olm-catalog/smart-gateway-operator/1.0.0/smartgateway.infra.watch_smartgateways_crd.yaml]: This is just the API version, and is unrelated to the versioning of the CSV :)

Comment (leifmadsen) [reply deploy/olm-catalog/smart-gateway-operator/1.0.0/smartgateway.infra.watch_smartgateways_crd.yaml]: ...or the operator container image tag
...or the Smart Gateway container image tag
...or the application registry version

Comment (leifmadsen) [reply deploy/olm-catalog/smart-gateway-operator/1.0.0/smartgateway.infra.watch_smartgateways_crd.yaml]: ...or the channel name

Comment (pkilambi) [reply deploy/olm-catalog/smart-gateway-operator/1.0.0/smartgateway.infra.watch_smartgateways_crd.yaml]: Thanks!

**Pull request merged**